### PR TITLE
article-nav

### PR DESCRIPTION
**Pull request by kailasnadh790**
Issue: https://github.com/hlxsites/famous-smoke-cigaradvisor/issues/20
Test URLs:
https://feature-article-nav--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor/posts/2023/12/top-10-best-new-cigars-of-2023

### CODE REVIEW

File: cigaradvisor/blocks/article-navigation/article-navigation.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { getAllPosts } from '../../scripts/scripts.js';
+
+function getFinalOrCurrentUrl(url) {
```
Comment (bstopp) [inline cigaradvisor/blocks/article-navigation/article-navigation.js:3]: What's this needed for?

Comment (kailasnadh790) [reply cigaradvisor/blocks/article-navigation/article-navigation.js:3]: article page will have a different url for imported articles , correct? which is 301 redirected to the sharepoint urls(present in the index) ?

Comment (kailasnadh790) [reply cigaradvisor/blocks/article-navigation/article-navigation.js]: removed

File: cigaradvisor/scripts/scripts.js [hunk 2]
```diff
@@ -229,6 +229,16 @@ export async function getPostByIdx(idx) {
...
 }
 
+/**
+ * Retrieves all posts and sorts them by published date.
+ * @param {boolean} sort - Indicates whether to sort the posts by published date in ascending order.
+ * @returns {Promise<Array>} - A promise that resolves to an array of post data.
+ */
+export async function getAllPosts() {
+  await loadPosts();
+  return articleIndexData;
```
Comment (bstopp) [inline cigaradvisor/scripts/scripts.js:239]: I'm now realizing that relying on the variable to be populated by insuring to call `loadPosts` doesn't make sense.

Can you update `loadPosts` to return the stored post list, and change `getPostByIndex` and `fetchPostsInfo` to not use the "global" variable?

**Review by bstopp — CHANGES_REQUESTED** [review]

File: cigaradvisor/scripts/scripts.js (hunk 2 above)
Comment (kailasnadh790) [reply cigaradvisor/scripts/scripts.js]: fixed

**Pull request merged**